prov/efa/src/efa_cq.h

```diff
@@ -3,6 +3,77 @@
 
 #include "efa.h"
 
+enum ibv_cq_ex_type {
+	IBV_CQ,
+	EFADV_CQ
+};
+
+struct efa_ibv_cq {
+	struct ibv_cq_ex *ibv_cq_ex;
+	enum ibv_cq_ex_type ibv_cq_ex_type;
+};
+
+struct efa_ibv_cq_poll_list_entry {
+	struct dlist_entry	entry;
+	struct efa_ibv_cq	*cq;
+};
+
+void efa_rdm_cq_poll_ibv_cq(ssize_t cqe_to_process, struct efa_ibv_cq *ibv_cq);
+
+static inline
+int efa_ibv_cq_poll_list_match(struct dlist_entry *entry, const void *cq)
+{
+	struct efa_ibv_cq_poll_list_entry *item;
+	item = container_of(entry, struct efa_ibv_cq_poll_list_entry, entry);
+	return (item->cq == cq);
+}
+
+
+static inline
+int efa_ibv_cq_poll_list_insert(struct dlist_entry *poll_list, struct ofi_genlock *lock, struct efa_ibv_cq *cq)
```

Comment:
Could move this logic to `ofi_list.h` as `dlist_insert_if_not_in_list`. The name of the function is `efa_ibv_cq_poll_list_insert` but the actual logic is inserting if the item is not in the list.

Reply:
We need to handle different type of data structure, I would make this logic a generalized macro and make both fid_list and efa_ibv_cq_poll_list onboard it.

Reply:
Would make this as a separate commit